pkg/segment/aggregations/segaggs_test.go

```diff
@@ -2605,93 +2605,113 @@ func Test_findSpan(t *testing.T) {
 
 }
 
+func WindowStreamStatsHelperTest(t *testing.T, values []float64, ssOption *structs.StreamStatsOptions, windowSize int, timestamps []uint64, measureFuncs []utils.AggregateFunctions, expectedValues [][]float64, expectedValues2 [][]float64, expectedLen [][]int, expectedSecondaryLen [][]int) {
+
+	for i, measureFunc := range measureFuncs {
+		ssResults := InitRunningStreamStatsResults(0)
+		fmt.Println(measureFunc)
+		for j, value := range values {
+			res, exist, err := PerformWindowStreamStatsOnSingleFunc(j, ssOption, ssResults, windowSize, measureFunc, value, timestamps[j], true)
+			assert.Nil(t, err)
+			if !ssOption.Current {
+				if j == 0 {
+					assert.False(t, exist)
+					assert.Equal(t, 0.0, res)
+				} else {
+					assert.True(t, exist)
+					assert.Equal(t, expectedValues[i][j-1], res)
+				}
+			} else {
+				assert.True(t, exist)
+				if measureFunc == utils.Avg {
+					assert.Equal(t, expectedValues[i][j], res)
+				}
+			}
+			assert.Equal(t, expectedLen[i][j], ssResults.Window.Len())
+			fmt.Println(expectedLen[i][j], ssResults.Window.Len())
+			if measureFunc == utils.Range {
+				assert.Equal(t, expectedSecondaryLen[i][j], ssResults.SecondaryWindow.Len())
+			}
+			if measureFunc == utils.Avg {
+				assert.Equal(t, expectedValues2[i][j], ssResults.CurrResult)
+			} else {
+				assert.Equal(t, expectedValues[i][j], ssResults.CurrResult)
+			}
+		}
+	}
+
+}
+
+func NoWindowStreamStatsHelperTest(t *testing.T, values []float64, ssOption *structs.StreamStatsOptions, measureFuncs []utils.AggregateFunctions, expectedValues [][]float64, expectedValues2 [][]float64, expectedValues3 [][]float64) {
+
+	for i, measureFunc := range measureFuncs {
+		ssResults := InitRunningStreamStatsResults(0)
+		ssOption.NumProcessedRecords = 0
+		if measureFunc == utils.Max {
+			ssResults = InitRunningStreamStatsResults(-math.MaxFloat64)
+		} else if measureFunc == utils.Min {
+			ssResults = InitRunningStreamStatsResults(math.MaxFloat64)
+		}
```

Comment:
I think this will be much cleaner if you change `InitRunningStreamStatsResults()` to take the aggregation function, and then internally it decides what to initialize the value to. Then you just need
```
ssResults := InitRunningStreamStatsResults(measureFunc)
ssOption.NumProcessedRecords = 0
```

Additionally, this will make it harder for developers to make a mistake when initializing it.